src/test/java/teammates/sqllogic/core/AccountRequestsLogicTest.java

```diff
@@ -29,6 +34,153 @@ public void setUpMethod() {
+    }
+
+    @Test
+    public void testDeleteAccountRequest_typicalRequest_success() {
+        AccountRequest ar = getTypicalAccountRequest();
+        when(accountRequestsDb.getAccountRequest(ar.getId())).thenReturn(ar);
+        accountRequestsLogic.deleteAccountRequest(ar.getId());
+
+        verify(accountRequestsDb, times(1)).deleteAccountRequest(any(AccountRequest.class));
+    }
+
+    @Test
+    public void testDeleteAccountRequest_nonexistentRequest_shouldSilentlyDelete() {
+        UUID nonexistentUuid = UUID.fromString("00000000-0000-4000-8000-000000000100");
+        accountRequestsLogic.deleteAccountRequest(nonexistentUuid);
+
+        verify(accountRequestsDb, times(1)).deleteAccountRequest(nullable(AccountRequest.class));
+    }
+
+    @Test
+    public void testGetAccountRequestByRegistrationKey_typicalRequest_success() {
+        AccountRequest ar = getTypicalAccountRequest();
+        String regkey = "regkey";
+        ar.setRegistrationKey(regkey);
+        when(accountRequestsDb.getAccountRequestByRegistrationKey(regkey)).thenReturn(ar);
+        AccountRequest actualAr =
+                accountRequestsLogic.getAccountRequestByRegistrationKey(ar.getRegistrationKey());
+
+        assertEquals(ar, actualAr);
+        verify(accountRequestsDb, times(1)).getAccountRequestByRegistrationKey(regkey);
+    }
+
+    @Test
+    public void testGetAccountRequestByRegistrationKey_nonexistentRequest_shouldReturnNull() throws Exception {
+        String nonexistentRegkey = "not_exist";
+        when(accountRequestsDb.getAccountRequestByRegistrationKey(nonexistentRegkey)).thenReturn(null);
+
+        assertNull(accountRequestsLogic.getAccountRequestByRegistrationKey(nonexistentRegkey));
+        verify(accountRequestsDb, times(1)).getAccountRequestByRegistrationKey(nonexistentRegkey);
+    }
+
+    @Test
+    public void testGetAccountRequest_typicalRequest_success() {
+        AccountRequest expectedAr = getTypicalAccountRequest();
+        when(accountRequestsDb.getAccountRequest(expectedAr.getEmail(), expectedAr.getInstitute())).thenReturn(expectedAr);
+        AccountRequest actualAr =
+                accountRequestsLogic.getAccountRequest(expectedAr.getEmail(), expectedAr.getInstitute());
+
+        assertEquals(expectedAr, actualAr);
+        verify(accountRequestsDb, times(1)).getAccountRequest(expectedAr.getEmail(), expectedAr.getInstitute());
+    }
+
+    @Test
+    public void testGetAccountRequest_nonexistentRequest_shouldReturnNull() {
+        String nonexistentEmail = "not-found@test.com";
+        String nonexistentInstitute = "not-found";
+        when(accountRequestsDb.getAccountRequest(nonexistentEmail, nonexistentInstitute)).thenReturn(null);
+
+        assertNull(accountRequestsLogic.getAccountRequest(nonexistentEmail, nonexistentInstitute));
+        verify(accountRequestsDb, times(1)).getAccountRequest(nonexistentEmail, nonexistentInstitute);
+    }
```

Comment:
Let's remove these. We will no longer be referencing by email address and institute. Also, removing is sufficient because there's no need to replace these to get by ID; it seems there are already tests for that below.